b2/console_tool.py

```diff
@@ -1040,6 +1042,48 @@ def _setup_logging(self, args, command, argv):
             logger.info('starting command [%s] with arguments: %s', command, argv)
 
 
+DATE_PATTERN = re.compile(r'^..., (\d\d) (...) (\d\d\d\d) (\d\d:\d\d\:\d\d) GMT$')
+
+MONTHS = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']
+
+
+class ClockSkewHook(HttpCallback):
+    def post_request(self, method, url, headers, http_response):
+        """
+        Raises an exception if the clock in the server is too different from the
+        clock on the local host.
+
+        The Date header contains a string that looks like: "Fri, 16 Dec 2016 20:52:30 GMT".
+        """
+        # Make a string that uses month numbers instead of month names
+        server_date_str = http_response.headers['Date']
+
+        # Convert the server time to a datetime object
+        try:
+            server_time = arrow.get(
+                server_date_str, 'ddd, DD MMM YYYY HH:mm:ss ZZZ'
+            )  # this, unlike datetime.datetime.strptime, always uses English locale
+        except arrow.parser.ParserError:
+            logger.exception('server returned date in an inappropriate format')
+            raise BadDateFormat(server_date_str)
+
+        # Get the local time
+        local_time = arrow.utcnow()
+
+        # Check the difference.  The timedelta.total_seconds() method is not available
+        # in Python 2.6, so we'll compute it using the formula from the Python docs.
+        max_allowed = 10 * 60  # ten minutes, in seconds
+        skew = local_time - server_time
+        skew_seconds = int(
+            (skew.microseconds + (skew.seconds + skew.days * 24 * 3600) * 1000000) / 1000000
+        )
+        if max_allowed < abs(skew_seconds):
+            if skew_seconds < 0:
+                raise ClockSkew('local clock is %d seconds behind server' % (-skew_seconds,))
```

Comment:
Formatting of this message should be in `b2.exceptions` as it is done for all other exceptions.